developer-docs-site/docs/tutorials/full-node/troubleshooting-fullnode.md

```diff
@@ -42,8 +42,18 @@ full_node_networks:
             addresses:
             - "/dns4/pfn0.node.devnet.aptoslabs.com/tcp/6182/ln-noise-ik/bb14af025d226288a3488b4433cf5cb54d6a710365a2d95ac6ffbd9b9198a86a/ln-handshake/0"
             role: "Upstream"
-
 ```
+## Incentivized Testnet Setup
+
+**Q: Generating only 1 key file instead of 3.**
+
+I am following [Run Validator Node Using Source Code](../validator-node/using-source-code.md) tutorial. One of my servers generates only 1 key file instead of 3, at step 7.
+
+**A:** Make sure you checkout the `testnet` branch as described in step 5.
+
+**Q: Do we need to run validator node and FullNode on two different machines, in order to get two different IPs?**
+
+**A:** We recommend that you run the validator node and the FullNode on two separate and independent machines that are well-provisioned and isolated from each other. Also, the FullNode IP is optional and you can leave it as blank. 
```

Comment:
I think this should be more front and center on the incentivized testnet or running a validator guide cc: @sherry-x

Reply:
Agree. I moved it to the introduction section of incentivized testnet docs.